src/App.tsx

```diff
@@ -1,24 +1,87 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { showError } from './helpers/helpers';
+import { Error } from './components/Error';
+import { Footer, FilterStatus } from './components/Footer';
+import { Todos } from './components/Todos';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 10881;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+
+  const [loader, setLoader] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => (showError('Failed to get list of todos', setError)));
+  }, []);
+
+  let visibleTodos = todos;
+
+  switch (filter) {
+    case FilterStatus.COMPLETED:
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+
+    case FilterStatus.ACTIVE:
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+
+    default:
+      break;
+  }
```

Comment:
1. You can rewrite this logic using only one filter method
2. I suggest you create separate function for this logic or put this code to useMemo